cmd/addaccount.go

```diff
@@ -61,79 +64,184 @@ func init() {
 }
 
 type AddAccountParams struct {
-	Entity
 	SignerParams
 	TimeParams
+	token    string
+	name     string
+	generate bool
+	keyPath  string
+	akp      nkeys.KeyPair
 }
 
 func (p *AddAccountParams) SetDefaults(ctx ActionCtx) error {
-	p.create = true
-	p.Entity.kind = nkeys.PrefixByteAccount
-	p.editFn = p.editAccount
+	p.generate = p.keyPath == ""
 	p.SignerParams.SetDefaults(nkeys.PrefixByteOperator, true, ctx)
 	return nil
 }
 
+func (p *AddAccountParams) resolveAccountNKey(s string) (nkeys.KeyPair, error) {
+	nk, err := store.ResolveKey(s)
+	if err != nil {
+		return nil, err
+	}
+	if nk == nil {
+		return nil, fmt.Errorf("a key is required")
+	}
+	t, err := store.KeyType(nk)
+	if err != nil {
+		return nil, err
+	}
+	if t != nkeys.PrefixByteAccount {
+		return nil, errors.New("specified key is not a valid account nkey")
+	}
+	return nk, nil
+}
+
+func (p *AddAccountParams) validateAccountNKey(s string) error {
+	_, err := p.resolveAccountNKey(s)
+	return err
+}
+
 func (p *AddAccountParams) PreInteractive(ctx ActionCtx) error {
 	var err error
-	if err = p.Entity.Edit(); err != nil {
+	p.name, err = cli.Prompt("account name", p.name, true, cli.LengthValidator(1))
+	if err != nil {
 		return err
 	}
 
-	if err = p.TimeParams.Edit(); err != nil {
+	p.generate, err = cli.PromptYN("generate an account nkey")
+	if err != nil {
 		return err
 	}
+	if !p.generate {
+		p.keyPath, err = cli.Prompt("path to an account nkey or nkey", p.keyPath, true, p.validateAccountNKey)
+		if err != nil {
+			return err
+		}
+	}
 
-	if err := p.SignerParams.Edit(ctx); err != nil {
+	if err = p.TimeParams.Edit(); err != nil {
 		return err
 	}
 
 	return nil
 }
 
 func (p *AddAccountParams) Load(ctx ActionCtx) error {
+	var err error
+	if p.generate {
+		p.akp, err = nkeys.CreateAccount()
```

Comment:
maybe in the future (think make an issue) combine these two steps so that you either get a stored key or an error. Maybe not worth it just musing.

Reply:
Only add account and add operator need the key a priory because they may be generating the key that signs them.